### PR TITLE
drop IE11 support

### PR DESCRIPTION
**Pull request by JoviDeCroock**
IE11 is coming to an [end on JUNE 15](https://www.lansweeper.com/eol/internet-explorer-11-end-of-life/)

- Removes Promise fallback
- Removes Symbol fallback
- Removes some DOM-specific IE impedements
- Removes the `removeNode` helper in favor of `Element.prototype.remove()`
- Removes the `assign` helper in favor of `Object.assign`

### CODE REVIEW

**Review by marvinhagemeister — APPROVED** [review]
Let's gooo 🙌🙌

File: src/component.js [hunk 1]
```diff
@@ -176,10 +175,7 @@ let rerenderQueue = [];
  */
 /* istanbul ignore next */
 // Note the following line isn't tree-shaken by rollup cuz of rollup/rollup#2566
-const defer =
-	typeof Promise == 'function'
-		? Promise.prototype.then.bind(Promise.resolve())
-		: setTimeout;
+const defer = Promise.prototype.then.bind(Promise.resolve());
```
Comment (porfirioribeiro) [inline src/component.js:178]: Wouldn't `queueMicrotask` also be a possibility here?

**Review by porfirioribeiro — APPROVED** [review]

File: src/component.js (hunk 1 above)
Comment (marvinhagemeister) [reply src/component.js:178]: Good catch. Browser support is looking really good for that too. What do you think @developit ?

Comment (developit) [reply src/component.js:178]: I had thought it was worse, but yeah you're right, it's decent. Main concern is Edge 16-18 and Safari 12, but those are looking pretty old these days...

**Comment by JoviDeCroock** [comment]
These changes are applied in https://github.com/preactjs/preact/pull/3589 and will hit main

**Pull request closed without merging**